### ISSUE TITLE
Python 3.9 breaks the package?

### ISSUE THREAD

**Issue by Boulder08** [yozik04/vallox_websocket_api#15]
I've installed Python v3.9 and after that, launching a scheduled Python script to collect data from the Vallox API fails with this error:
```

File "C:\Program Files\Python39\lib\site-packages\vallox_websocket_api\__init__.py", line 1, in <module>
    from .client import Client
  File "C:\Program Files\Python39\lib\site-packages\vallox_websocket_api\client.py", line 7, in <module>
    from .messages import (LogReadRequest, LogReadResponse1, LogReadResponse2,
  File "C:\Program Files\Python39\lib\site-packages\vallox_websocket_api\messages.py", line 91, in <module>
    LogReadResponse2 = GreedyRange(
  File "C:\Program Files\Python39\lib\site-packages\construct\core.py", line 444, in compile
    module = importlib.util.module_from_spec(module_spec)

AttributeError: module 'importlib' has no attribute 'util'
```

This is the code I've tested it with:
```

# -*- coding: utf-8 -*-
import asyncio
import datetime
from vallox_websocket_api import Client

client = Client('192.168.0.60')

async def run():
    metrics = await client.fetch_metrics()
    poimi_muuttujat(metrics)

def poimi_muuttujat(metrics):
    global poistoilma
    global jateilma
    global ulkoilma
    global tuloilmakennosta
    global tuloilmakennosta_heater
    global kennontila
    global tulokierrokset
    global poistokierrokset
    global puhallin
    poistoilma = metrics['A_CYC_TEMP_EXTRACT_AIR']
    jateilma = metrics['A_CYC_TEMP_EXHAUST_AIR']
    ulkoilma = metrics['A_CYC_TEMP_OUTDOOR_AIR']
    tuloilmakennosta = metrics['A_CYC_TEMP_SUPPLY_CELL_AIR']
    tuloilmakennosta_heater = metrics['A_CYC_TEMP_SUPPLY_AIR']
    kennontila = metrics['A_CYC_CELL_STATE']
    tulokierrokset = metrics['A_CYC_SUPP_FAN_SPEED']
    poistokierrokset = metrics['A_CYC_EXTR_FAN_SPEED']
    puhallin = metrics['A_CYC_FAN_SPEED']

asyncio.get_event_loop().run_until_complete(run())
```

It's strange that if I open the script in IDLE and launch it from there, no problems.
All the old Python versions are uninstalled.

**Comment by tirkarthi** [comment]
See https://github.com/construct/construct/pull/930 PR to construct. This seems to be a behavior change in Python's importlib. See also https://issuetracker.google.com/issues/170125513

**Comment by Boulder08** [comment]
This issue is fixed by adding
`import importlib.util`
at the beginning of the messages.py file.

**Comment by yozik04** [comment]
> This issue is fixed by adding
> `import importlib.util`
> at the beginning of the messages.py file.

If this is the fix then this import can be anywhere? Not only in vallox/construct libs?

**Comment by Boulder08** [comment]
It's v2.10.67.

**Comment by Boulder08** [comment]
> > This issue is fixed by adding
> > `import importlib.util`
> > at the beginning of the messages.py file.
> 
> If this is the fix then this import can be anywhere? Not only in vallox/construct libs?

I'm not savvy at all, but I'd figure that it's not global so if you use it inside the module, you have to import it there.

**Comment by slovdahl** [comment]
I'm also having the same problem, and it makes no sense at all. I'm on macOS 11.6.3 at the moment and trying to run a simple test client with `vallox_websocket_api` in iTerm gives me the same `module 'importlib' has no attribute 'util'` error. I have tested it with in Python 3.10.2, 3.9.10 and 3.8.12, and I get the same error in all three cases.

```python
import asyncio
from vallox_websocket_api import Vallox

client = Vallox('1.2.3.4')

async def run():
    profile = await client.get_profile()
    print("Profile: %s" % profile)

asyncio.get_event_loop().run_until_complete(run())
```

If a manually add the changes from https://github.com/construct/construct/pull/930 to `construct` in `site-packages`, it works fine.

**Comment by yozik04** [comment]
I can not reproduce. I test the lib in 5 different python versions: `["3.6", "3.7", "3.8", "3.9", "3.10"]`.
I do not get any errors. Maybe construct version must be stricter.
```
install_requires=[
        "websockets >= 9.1, < 11.0",
        "construct >= 2.9.0, < 3.0.0"
],
```

**Comment by slovdahl** [comment]
My guess is that it has somthing to do with either the OS, the surrounding environment or how the Python binary/package is built/packaged. Did you try it with the minimal example I provided too?

**Comment by yozik04** [comment]
Can you run tests?
`python3 setup.py test`

I run the lib on Ubuntu server in venv 3.8.10 to hook the unit up with MQTT.
I run tests on Mac Big Sur with Python 3.9 and all is green.

**Comment by yozik04** [comment]
On my Mac tests do work in IDE (PyCharm) and from console.

**Comment by slovdahl** [comment]
@yozik04 what about this simple test script, does it also work on your Mac? And how is Python installed in that case?

```python
import asyncio
from vallox_websocket_api import Vallox

client = Vallox('1.2.3.4')

async def run():
    profile = await client.get_profile()
    print("Profile: %s" % profile)

asyncio.get_event_loop().run_until_complete(run())
```

**Comment by yozik04** [comment]
Works fine:
```
python3 slovdahl.py
Profile: PROFILE.HOME
```

**Comment by yozik04** [comment]
websockets==9.1
construct==2.10.67

**Comment by yozik04** [comment]
@slovdahl, @Boulder08 I have asked construct devs to merge PR to fix this problem. Please verify. Should be fixed in construct 2.10.68.

**Comment by slovdahl** [comment]
> @slovdahl, @Boulder08 I have asked construct devs to merge PR to fix this problem. Please verify. Should be fixed in construct 2.10.68.

Great, I can confirm that it starts working after running `pip3 install construct==2.10.68`. 👍

**Comment by yozik04** [comment]
Great! Happy to help.
